### PR TITLE
FE-813 Update Checkbox and Checkbox.Group Components

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
Restyles Checkbox
- #355 - Checkbox `id` prop is now required
- #355 - Checkbox now properly hides a screen reader accessible label
- #355 - Checkbox now properly uses aria-describedby to link to its helptext and error

Screen Shot 2020-03-17 at 3 07 16 PM

### How To Test or Verify
- Run storybook with `npm run start:storybook`
- Visit & verify Checkbox stories

### PR Checklist
- [x] Add your changes to `unreleased.md` in the root directory. If you plan on publishing immediately after merging (such as a hotfix) or aren't making changes to a published package, you can ignore this step.
- [x] Provide screenshots or [screen recordings](https://getkap.co/) for any visual changes.
- [x] Get approval from a UX team member (**#uxfe** or **#design-guild** on Slack) for any visual changes.

### CODE REVIEW

File: packages/matchbox/src/components/Checkbox/styles.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { visuallyHidden } from '../../styles/helpers';
+import styled from 'styled-components';
+import { margin } from 'styled-system';
+
+export const Wrapper = styled('div')`
+  ${margin}
+`;
+
+export const StyledBox = styled(Box)`
+  border: 2px solid ${props => (props.error ? tokens.color_red_700 : tokens.color_gray_500)};
```
Comment (jonambas) [inline packages/matchbox/src/components/Checkbox/styles.js]: Don't think this gray border is accessible, same with some of the other grays in the disabled styles – will ask UX team to check it out

File: packages/matchbox/src/components/Checkbox/tests/Checkbox.test.js [hunk 2]
```diff
@@ -1,45 +1,91 @@
...
+      modifier: box,
+    });
+    expect(wrapper.text()).toEqual('test-error');
+    expect(wrapper.find('input')).toHaveAttributeValue('aria-describedby', 'test-id-error');
+    expect(wrapper.find('div').at(2)).toHaveAttributeValue('id', 'test-id-error');
+  });
+
+  it('renders disabled', () => {
+    const wrapper = subject({ disabled: true });
+    expect(wrapper.find('label')).toHaveStyleRule('cursor', 'not-allowed', { modifier: ':hover' });
```
Comment (nicklemmon) [inline packages/matchbox/src/components/Checkbox/tests/Checkbox.test.js]: Could you check for the `disabled` attribute on the `<input/>`?

File: packages/matchbox/src/components/Checkbox/tests/Checkbox.test.js [hunk 3]
```diff
@@ -1,45 +1,91 @@
...
+  });
 
+  it('renders required', () => {
+    const wrapper = subject({ required: true, label: 'test-label' });
+    expect(wrapper.text()).toEqual('test-label*');
```
Comment (nicklemmon) [inline packages/matchbox/src/components/Checkbox/tests/Checkbox.test.js]: Same as above, could you check for the `required` attribute on the `<input/>`?

File: packages/matchbox/src/components/ThemeProvider/globalStyles.js [hunk 4]
```diff
@@ -112,6 +112,7 @@ export default `
   a,
   a:visited {
     color: ${tokens.color_blue_700};
+    text-decoration: underline;
```
Comment (nicklemmon) [inline packages/matchbox/src/components/ThemeProvider/globalStyles.js]: Thanks for doing this - we'll end up removing the underline on a case-by-case basis in the app, but having this as the default is the right thing to do.

**Pull request merged**